org.sonarlint.eclipse.core.tests/src/test/java/org/sonarlint/eclipse/core/internal/telemetry/SonarLintTelemetryTest.java

```diff
@@ -157,4 +188,67 @@ public void analysisDoneOnSingleFile_should_not_trigger_analysisDoneOnSingleFile
     verify(engine).isEnabled();
     verifyNoMoreInteractions(engine);
   }
+
+  @Test
+  public void should_not_report_sonar_cloud_usage_when_project_is_not_bound() {
+    clearImportedProjectBinding();
+
+    assertThat(SonarLintTelemetry.isAnyOpenProjectBoundToSonarCloud()).isFalse();
+  }
+
+  @Test
+  public void should_not_report_sonar_cloud_usage_when_project_is_bound_to_localhost() {
+    addServer("localhost", "http://localhost:9000");
+    bindImportedProjectToServer("localhost");
+
+    assertThat(SonarLintTelemetry.isAnyOpenProjectBoundToSonarCloud()).isFalse();
+  }
+
+  @Test
+  public void should_report_sonar_cloud_usage_when_project_is_bound_to_sonar_cloud() {
```

Comment:
To me one test is missing, the one at the source of the ticket: `should_not_report_sonar_cloud_usage_when_sonarcloud_connection_but_project_is_not_bound`

Reply:
Interesting ! It was deliberate. But in the end it might be better to have it